includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -380,13 +385,13 @@ public function create_setup_intent() {
 
 		$enabled_payment_methods = array_filter( $this->get_upe_enabled_payment_method_ids(), [ $this, 'is_enabled_for_saved_payments' ] );
 
-		$setup_intent = $this->payments_api_client->create_setup_intention(
-			$customer_id,
-			array_values( $enabled_payment_methods )
-		);
+		$request = Create_Setup_Intention::create();
+		$request->set_customer( $customer_id );
+		$request->set_payment_method_types( array_values( $enabled_payment_methods ) );
+		$setup_intent = $request->send( 'wcpay_create_setup_intention_request' );
 		return [
-			'id'            => $setup_intent['id'],
-			'client_secret' => $setup_intent['client_secret'],
+			'id'            => $setup_intent->offsetGet( 'id' ),
```

Comment:
The idea of implementing `ArrayAccess` was to allow this line to just be `$setup_intent['id']`.

Reply:
Yeah... I totally missed it. It is fixex.